app/models/game.rb

```diff
@@ -32,7 +32,8 @@ def check(id, opponent_id)
 
   def determine_checkmate
     return checkmate(white_player_id) if turn_number.even? && determine_check
-    checkmate(black_player_id) if turn_number.odd? && determine_check
+    return checkmate(black_player_id) if turn_number.odd? && determine_check
+    false
```

Comment:
1. I updated the method move_to! and the controller so now the user sees an alert message preventing them moving onto check.
2. One test of checkmate didn't pass - I commented it, because Jonathan worked on that issue.
3. I found a bug relating to an instance variable. The variable @piece in 'try_to_move' changed in the middle of the method, after calling 'castling'. Since it had a different value in 'castling'.
I changed the instance variable @piece to a local one (piece), in those two methods.